BayeuxClient.agent.lib.nut

```diff
@@ -0,0 +1,936 @@
+            callback(null);
+            return;
+        }
+        local errMsg = null;
+
+        if (success) {
+            // "clientId" NOT in response
+            errMsg = "Handshake response does not have \"clientId\" field!";
+            _logError(errMsg);
+        } else if ("error" in response) {
+            errMsg = response["error"];
+        } else {
+            errMsg = "Handshake was unsuccessful, but response does not have \"error\" field!";
+            _logError(errMsg);
+        }
+        callback(Bayeux.Error(BAYEUX_CLIENT_ERROR_TYPE.BAYEUX_ERROR, errMsg));
+    }
+
+    function _connect(establish = false) {
+        _reconnectTimer = null;
+
+        local msgId = _nextMessageId();
+        local message = {
+            "id" : msgId,
+            "clientId" : _clientId,
+            "channel" : BAYEUX_CLIENT_META_CHANNEL_CONNECT,
+            "connectionType" : BAYEUX_CLIENT_LONGPOLL_TRANSPORT_NAME,
+            "advice" : {
+                "timeout" : establish ? 0 : _advice.timeout
+            }
+        }
+
+        local onResponse = _onReconnectResponse;
+
+        if (establish) {
+            onResponse = _onConnectResponse;
+        }
+
+        _metaHandlers[msgId] <- onResponse.bindenv(this);
+
+        local sent = function(error) {
+            if (error != null) {
+                if (msgId in _metaHandlers) {
+                    delete _metaHandlers[msgId];                
+                }
+                if (establish) {
+                    _onConnected(error);
+                } else {
+                    _onDisconnected(error);
+                }
+            }
+        }.bindenv(this);
+
+        _send(message, sent, true);
+    }
+
+    function _onConnectResponse(response, establish = true) {
+        _log("Connect response received:");
+        _logTable(response);
+
+        local callback = null;
+
+        if (establish) {
+            // Connection is being established now
+            callback = _onConnected;
+        } else {
+            // Connection has been established and now we are maintaining it
+            // In case of any error we will call _onDisconnected
+            callback = _onDisconnected;
+        }
+
+        if (!_checkSuccessfulField(response, callback)) {
+            return;
+        }
+
+        if (response["successful"]) {
+            if (!_isDisconnecting) {
+                _delayedReconnect(_advice.interval);
+            }
+            if (establish) {
+                callback(null);
+            }
+            return;
+        } 
+
+        local reconnectAdvice = null;
+        if ("advice" in response && "reconnect" in response["advice"]) {
+            reconnectAdvice = response["advice"]["reconnect"]
+        }
+
+        if (reconnectAdvice == "retry") {
+            // We can reconnect automatically
+            if (!_isDisconnecting) {
+                // _backoff is in secs, so we need to convert it to msecs
```

Comment:
NO! `_delayedReconnect` expects seconds!!!

Please let's keep everything in seconds internally.

Reply:
Please add a test for this.

Reply:
Done